Refaktoring komponentów CourseFilter oraz CourseList: Analiza uzycia komponentow CourseFilter oraz CourseList. Dodanie wyjasniajacych komentarzy w tych komponentach. Refaktoring komponentow CourseList w celu zmniejszenia liczby widocznych roznic. Zmiany dotyczą głównie ujednolicenia sposobu deklaracji komponentów Vue i są nieinwazyjne. Porządki w importach dotyczących wyżej wymienionych komponentów.

rafalstarypan opened the pull request.

rafalstarypan committed:
Remove unused imports. Add clarifying commments

rafalstarypan committed:
Fix formatting

rafalstarypan committed:
Refactor CourseList component

rafalstarypan commented:
Ten PR zawiera nastepujace zmiany:
1. Analiza uzycia komponentow CourseFilter oraz CourseList. Dodanie wyjasniajacych komentarzy w tych komponentach.
2. Refaktoring komponentow CourseList w celu zmniejszenia liczby widocznych roznic. Zmiany dotyczą głównie ujednolicenia sposobu deklaracji komponentów Vue i są nieinwazyjne.
3. Porządki w importach dotyczących wyżej wymienionych komponentów.

jakonradk committed:
Poprawka błędu po refaktoryzacji komponentu CourseList.vue

Po refaktoryzacji komponentu CourseList.vue występował błąd powodujący brak funkcjonalności checkboxów na liście przedmiotów.
Błąd występował, jak rozumiem, z powodu definiowania property "selection" w elemencie "methods", który służy do definiowania funkcji, przez co komendy "click="selection = []" oraz "v-model:selection" nie mogły zbindować się do property "selection".
Aby nie modyfikować tagu "<template>" przeniosłem definicję property "selection" do elementu "computed". Property w tym elemencie będą automatycznie przeliczane, gdy zmienią się ich zależności (w przypadku "selection" będzie to "this.selectionState"), ale zmiany w this.selectionState są przewidywalne i zgodnie z plikiem "../store/courses.ts" następują wyłącznie po wywołaniu settera "selection", więc nie powinno to powodować niechcianych zachowań.

lgpawel commented on zapisy/apps/enrollment/timetable/assets/components/CourseList.vue:
```diff
@@ -25,6 +25,16 @@ export default Vue.extend({
     }),
+    selection: {
+		  get(): number[] {
+        console.log("getter");
```
Oprócz ugłaskania lintera proszę jeszcze usunąć `console.log`, na pewno docelowo nie są pożądane.

jakonradk committed:
Lint + usunięcie console.log()

jakonradk committed:
Lint jeszcze raz

jakonradk committed:
Ujednolicenie komentarzy dotyczących miejsc używania komponentów

Komentarze zostały zamienione na takie, które wskazują odpowiedni webpack, w którym dany komponent jest używany.

jakonradk committed:
Dodanie komentarzy wyjaśniających różnice między komponentami CourseList w innych miejscach

Komentarze mają na celu wyjaśnienie odpowiednich unikalnych funkcji w każdym z komponentów i zwrócenie uwagi na niejasności.

jakonradk committed:
Dodanie komentarzy wyjaśniającego różnice w komponencie CourseFilter na stronie głosowania

Komentarze mają na celu wyjaśnienie unikalnej zawartości w komponencie CourseFilter na stronie głosowania w porównaniu do innych komponentów o tej samej nazwie.

lgpawel commented on zapisy/apps/offer/vote/assets/components/CourseFilter.vue:
```diff
@@ -13,6 +13,7 @@ import {
   MultiselectFilterData,
 } from "@/enrollment/timetable/assets/models";
 
+// This component is used in the vote-point-counter webpack
```
Komentarz dałbym zresztą na samym początku pliku czy tam po znaczniku `<script>` – w niektórych innych komponentach z tego PRa już tam są jakieś inne ogólne komentarze i zwłaszcza w tamtych przypadkach nie ma sensu rozwlekać tego po kilku miejscach w kodzie.

lgpawel commented on zapisy/apps/offer/vote/assets/components/CourseFilter.vue:
```diff
@@ -76,6 +77,8 @@ export default Vue.extend({
     }
 
+    // this fragment is responsible for
+    // handling changes in the state
```
Analogiczny komentarz przydałby się w `timetable/.../CourseList.vue`, razem z informacją. **co** właściwie jest w _state_ (tam: ptaszki postawione przez użytkownika przy przedmiotach, tu: nie pamiętam).

jakonradk committed:
webpack -> asset, przeniesienie informacji o miejscu używania na górę pliku

lgpawel committed:
I don't like this one

lgpawel committed:
Yet another *crucial* uniformization

lgpawel committed:
Ma, I'm a serious developer
